python/lsst/pipe/tasks/maskStreaks.py

```diff
@@ -407,11 +408,18 @@ def line_search(c, dx):
             if chi2 == 0:
                 break
             if not np.isfinite(A).all():
-                # TODO: DM-30797 Add warning here.
                 fitFailure = True
+                if log is not None:
```

Comment:
Ah so THIS is the canonical way to log inside a function like this? Fascinating. (In what weird situation would log be None??)

Reply:
I'm not sure, but it looks like that there are a bunch of other places in the stack where the logger is passed around like this. I made it optional because it's possible we would want to use the line profile fitter outside the context of task, in which case you wouldn't have a logger set up.